keyserver/server.go

```diff
@@ -193,10 +192,15 @@ func (server *ConiksServer) Run(addrs []*Address) {
 				Handler:   mux,
 				TLSConfig: tlsConfig,
 			}
-			server.httpServer = httpSrv
 			go func() {
 				httpSrv.Serve(ln)
 			}()
+			go func() {
```

Comment:
Please add `server.waitStop.Add(1)` before this routine and `server.waitStop.Done()` after `httpSrv.Shutdown()`.